aeron-client/src/main/java/io/aeron/protocol/DataHeaderFlyweight.java

```diff
@@ -156,7 +166,18 @@ public static boolean isHeartbeat(final UnsafeBuffer packet, final int length)
      */
     public static boolean isEndOfStream(final UnsafeBuffer packet)
     {
-        return BEGIN_END_AND_EOS_FLAGS == (packet.getByte(FLAGS_FIELD_OFFSET) & 0xFF);
+        return (packet.getByte(FLAGS_FIELD_OFFSET) & EOS_FLAG) != 0;
```

Comment:
It's worth pointing out that this change means that old java drivers won't detect EOS or REVOKED when they see a REVOKE/EOS from a newer driver.

Reply:
This change is being pushed to the 1.47.x branch for backwards compatibility issues, should they arise.  Otherwise we're just going to document the fact that revoke() shouldn't be used against < 1.48 _Java_ drivers.